src/reducers/versions.tsx

```diff
@@ -666,6 +666,10 @@ const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
   switch (action.type) {
     case getType(actions.loadVersionInfo): {
       const { version } = action.payload;
+      if (state.versionInfo[version.id]) {
+        // We shouldn't load the same version again.
+        return state;
```

Comment:
I know that this solves the problem but this isn't the right place to do it. What if we need to re-fetch the same version from the API because it changed on the server? That would no longer be possible.

What about not dispatching a version in `fetchDiff()` when it already exists? https://github.com/mozilla/addons-code-manager/blob/fcac49f0babb33fa301522e932dd969343c0957c/src/reducers/versions.tsx#L621

I still think it's weird that `fetchDiff()` loads a version when switching paths. Maybe we need a new action, like `fetchDiffForVersion()`, that we can call when we know we are only fetching a diff for the same version.

Reply:
The reason this isn't happening in `Browse` is because we already have two different actions.

We have `fetchVersion()`:

https://github.com/mozilla/addons-code-manager/blob/fcac49f0babb33fa301522e932dd969343c0957c/src/pages/Browse/index.tsx#L92

And we have `fetchVersionFile()`:

https://github.com/mozilla/addons-code-manager/blob/fcac49f0babb33fa301522e932dd969343c0957c/src/pages/Browse/index.tsx#L114

Reply:
It does seem like redesigning the way `Compare` works to be more like `Browse` might make sense, and I started looking at that, but it was becoming a bigger patch. I think that making the change you suggest above, to just not dispatch `loadVersionInfo` from inside `fetchDiff` if the version is already loaded is also a decent solution to the immediate problem and much simpler, so thats what I've done.